src/test/java/org/apache/commons/lang3/StringUtilsChompTest.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.lang3;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.infra.Blackhole;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.commons.lang3.StringUtils.EMPTY;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.apache.commons.lang3.StringUtils.length;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+public class StringUtilsChompTest {
+    private static final String string1 = "\r";
+    private static final String string2 = "a";
+    private static final String string3 = "aa";
+    private static final String string4 = "aa\r";
+    private static final String string5 = "aa\n";
+    private static final String string6 = buildString6();
```

Comment:
string6 will be the same as string5

Reply:
> string6 will be the same as string5

my bad. I thought string5 would be "\r\n" and 6 be "\n", maybe I made a mistake.

Reply:
```java
// Different code paths with no chomp
private static final String string1 = "";
private static final String string2 = "a";
private static final String string3 = "aa";

// Single char chomp
private static final String string4 = "\r";
private static final String string5 = "\n";

// Multi-char chomp
private static final String string6 = "\r\n";
private static final String string7 = "a\n";
private static final String string8 = "a\r";
private static final String string9 = "a\r\n";
```
That should be all code paths.